### PR TITLE
Lint and update nix files

### PR DESCRIPTION
**Pull request by peter-mlabs**
- add `hlint` to the precommit
- "adds" `apply-refact` to the dev shell, (but actually comments it out because it is broken)
- lints the codebase
- updates the `.envrc` to track the various nix files

### CODE REVIEW

File: Plutarch/Api/V1/AssocMap.hs [hunk 1]
```diff
@@ -714,6 +717,7 @@ zipMergeInsertCommutative (MergeHandlerCommutative bothPresent onePresent) = unT
           PassOne -> pcons # x # xs'
           HandleOne handler ->
             List.pmap
+              {- HLINT ignore "Redundant bracket" -}
```
Comment (bladyjoker) [inline Plutarch/Api/V1/AssocMap.hs:720]: This is generally pretty safe, was there a problem with hls applied solution? Fyi, I usually visit this file and HLS provides an action that refactors the thing. I never had a semantic issue with it.

File: flake.nix [hunk 2]
```diff
@@ -61,6 +61,14 @@
                 project.hsPkgs.hspec-discover.components.exes."hspec-discover"
                 project.hsPkgs.markdown-unlit.components.exes."markdown-unlit"
               ];
+              #   FIXME (peter-mlabs): This is currently broken.
```
Comment (bladyjoker) [inline flake.nix:64]: nit; Syntax is `FIXME(peter-mlabs): This is currently broken.` (without the space)

Also FIXME would indicate there's a behavior in need of fixing. Perhaps TODO is more appropriate? As we want to add a new behavior.

File: flake.nix [hunk 3]
```diff
@@ -61,6 +61,14 @@
...
+              #   In the future you can do something like
+              #
+              #     find -name '*.hs' -not -path './dist-*/*' -exec hlint -j --refactor --refactor-options="--inplace" {} +
```
Comment (bladyjoker) [inline flake.nix:67]: well nice! We can make a pre-commit-hooks module that does that. Not sure if it's safe to do such refactors.

**Review by bladyjoker — APPROVED** [review]
Thanks @peter-mlabs

File: flake.nix (hunk 2 above)
Comment (peter-mlabs) [reply flake.nix:64]: Wait, there's standard syntax around this?? I've always done this informally! 

What is the spec for this?

Comment (bladyjoker) [reply flake.nix:64]: Lol spec! The only thing close to that I've seen was https://peps.python.org/pep-0350/.

I'm hoping to adopt things like https://github.com/pgilad/leasot and have up to date reports with codetags.

`<TODO|FIX|WARN|HACK|WHATEVER>(<username|bug link>): <Description>`

That's what I've seen used in bigger codebases. Emacs magit-todos for example is excellent in representing this.

**Pull request merged**